src/components/sbb-icon/sbb-icon.tsx

```diff
@@ -0,0 +1,185 @@
+import { Component, Element, h, Host, Prop, State, Watch } from '@stencil/core';
+import { getSvgContent, iconNamespaces, cachedIcons, registeredIcons } from './request';
+import { validateContent } from './validate';
+
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-icon.scss',
+  tag: 'sbb-icon',
+})
+export class SbbIcon {
+  private _svgName: string;
+  private _svgNamespace = 'sbb';
+  private _ariaHidden: string;
+
+  @Element() private _element!: HTMLElement;
+
+  /**
+   * The icon svg content rendered on the page: <svg>...</svg>.
+   */
+  @State() private _svgIcon: string;
+
+  /**
+   * Only set the aria-label if aria-hidden is not set to "true".
+   */
+  @State() private _ariaLabel: string;
+
+  /**
+   * The provided name consisting of the namespace and the name of the icon.
+   * If the namespace is missing, the default namespace "sbb" will be used.
+   * E.g. `name` (will use "sbb" as namespace) or `namespace:name`.
+   */
+  @Prop() public name: string;
+
+  /**
+   * When set to `false`, SVG content that is HTTP fetched will not be checked
+   * if the response SVG content has any `<script>` elements, or any attributes
+   * that start with `on`, such as `onclick`.
+   * @default true
+   */
+  @Prop() public sanitize = true;
+
+  public connectedCallback(): void {
+    this._loadSvgIcon(this.name);
+  }
+
+  /**
+   * Update the icon svg content on name change
+   */
+  @Watch('name')
+  public updateSvgIcon(newValue: string, oldValue: string): void {
+    if (newValue !== oldValue) {
+      this._element.removeAttribute('aria-label');
+      this._loadSvgIcon(newValue);
+    }
+  }
+
+  private _loadSvgIcon(iconName: string): void {
+    if (!iconName) {
+      return;
+    }
+
+    const [namespace, name] = this._splitIconName(iconName);
+
+    if (namespace) {
+      this._svgNamespace = namespace;
+    }
+
+    if (name) {
+      this._svgName = name;
+    }
+
+    this._ariaHidden = this._getAriaHidden();
+
+    // generate a default label in case user does not provide their own
+    if (this._ariaHidden === 'false' && this._svgName) {
+      this._ariaLabel = this._getAriaLabel() || `Icon ${this._svgName.replace(/-/g, ' ')}`;
+    }
+
+    // try to load SVG from registered icons
+    this._svgIcon = this._loadRegisteredIcon();
+
+    if (this._svgIcon) {
+      return;
+    }
+
+    const url = this._resolveSvgUrl();
+
+    if (url) {
+      if (cachedIcons.has(url)) {
+        // if it's already loaded
+        this._svgIcon = cachedIcons.get(url);
+      } else {
+        getSvgContent(url, this.sanitize).then(() => (this._svgIcon = cachedIcons.get(url)));
```

Comment:
See consideration in request.ts. That change would allow the following usage:
```suggestion
        this._svgIcon = await getSvgContent(url, this.sanitize);
```
What are your thoughts?

Reply:
Good suggestion, I like it. Much cleaner, thank you very much! 👍